src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,132 @@
+    private Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
+    public void put(K key, V value) {
+        if (key == null) {
+            resize();
+            int index = 0;
+            Node<K, V> node = buckets[index];
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+            buckets[index] = new Node<>(key, value, buckets[index]);
+            size++;
+        } else {
+            int index = getIndex(key);
+            Node<K, V> newNode = new Node<>(key, value, null);
+            if (buckets[index] == null) {
+                buckets[index] = newNode;
+                size++;
+            } else {
+                Node<K, V> currentNode = buckets[index];
+                Node<K, V> prevNode = null;
+                while (currentNode != null) {
+                    if (Objects.equals(key, currentNode.key)) {
+                        currentNode.value = value;
+                        return;
+                    }
+                    prevNode = currentNode;
+                    currentNode = currentNode.next;
+                }
+                if (prevNode != null) {
+                    Node<K, V> newNextNode = prevNode.next;
+                    prevNode.next = newNode;
+                    newNode.next = newNextNode;
+                    size++;
+                }
+            }
+            if ((float) size / buckets.length >= LOAD_FACTOR) {
+                resize();
+            }
+        }
     }
 
-    @Override
     public V getValue(K key) {
+        for (Node<K, V> node : buckets) {
+            while (node != null) {
+                if (Objects.equals(node.key, key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
-    @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        int index = (key == null) ? 0 : (key.hashCode() % buckets.length);
+        return index > 0 ? index : Math.abs(index);
+    }
+
+    private void resize() {
+        if ((float) size / buckets.length >= LOAD_FACTOR) {
+            int newCapacity = buckets.length * 2;
+            Node<K, V>[] newBuckets = new Node[newCapacity];
+            int newSize = 0;
+
+            for (Node<K, V> node : buckets) {
+                while (node != null) {
+                    Node<K, V> nextNode = node.next;
+                    int newIndex = getIndex(node.key);
+                    node.next = newBuckets[newIndex];
+                    newBuckets[newIndex] = node;
+                    node = nextNode;
+                    newSize++;
+                }
+            }
+
+            buckets = newBuckets;
+            size = newSize;
```

Comment:
code is identical to lines 102-117
please use if/else to calculate newCapacity and then put all remaining code once after if/else block